File: linked list/add two number.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
//     we are given with two non empty linked list 
//     two non neg integer
//     stored in reversed order
    
    
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
//         using dummy
        // temp = dummy
        
        ListNode *dummy =  new ListNode();
        ListNode *temp = dummy;
        int carry =0;
        // now usinh while to check 
        
        while(l1!=NULL || l2!=NULL ||carry){
        int sum =0;
//             now checking the list 
        
        if(l1 != NULL)
        {
            sum+= l1->val;
            l1 = l1->next;
        }
          if(l2 != NULL)
        {
            sum+= l2->val;
            l2 = l2->next;
        }
        
        sum+=carry;
//             carry sum divied by 10
            carry = sum/10;
//             now checking it
        ListNode *node = new ListNode(sum%10);
        temp->next= node;
        temp =temp->next;
        
    }
    return dummy->next;
}
};

File: linked list/add two number.py
# python solution

class Solution:
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        v1 = l1.val;    c1 = 1
        v2 = l2.val;    c2 = 1
        while(l1.next != None):
            v1 = v1 + l1.next.val * (10 ** c1)
            l1 = l1.next
            c1 += 1
        while(l2.next != None):
            v2 = v2 + l2.next.val * (10 ** c2)
            l2 = l2.next
            c2 += 1
        s = v1 + v2

        s = str(s)
        res = None
        for i in s:
            node = ListNode(i[::-1])
            node.next = res
            res = node

        return res  
